File: Android/app/src/main/java/com/bdreiss/trackmyattack/listeners/CustomListener.java
package com.bdreiss.trackmyattack.listeners;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.view.View;

import main.java.com.bdreiss.dataAPI.DataModel;
import main.java.com.bdreiss.dataAPI.Intensity;

public class CustomListener implements View.OnClickListener {

    protected  Context context;
    protected DataModel data;
    protected String text;

    CustomListener(Context context, DataModel data){
        this.context = context;
        this.data = data;
    }



    public void setTextValue(String text){
        this.text = text;
    }

    public CustomListener copy(){
        CustomListener newListener = new CustomListener(context, data);
        newListener.setTextValue(text);
        return newListener;
    }

    public DataModel getData(){
        return data;
    }

    public void chooseIntensity(Context context, DialogInterface.OnClickListener listener){
        String[] intensities = new String[Intensity.values().length-1];

        for (int i=1;i <Intensity.values().length;i++)
            intensities[i-1] = Intensity.values()[i].toString();

        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle("Choose Intensity");
        builder.setItems(intensities, listener);
        builder.setNegativeButton("Cancel", null);
        builder.create();
        builder.show();
    }

    @Override
    public void onClick(View view) {

    }
}

File: Android/app/src/main/java/com/bdreiss/trackmyattack/listeners/CustomListener.kt
package com.bdreiss.trackmyattack.listeners

import android.content.Context
import android.view.View
import androidx.appcompat.app.AlertDialog
import main.java.com.bdreiss.dataAPI.DataModel
import main.java.com.bdreiss.dataAPI.Intensity

//abstract class for listeners being passed to layouts showing causes, symptoms and remedies
abstract class CustomListener(val context: Context, val data: DataModel) : View.OnClickListener{
    lateinit var text : String

    fun setTextValue(text : String){
        this.text = text
    }

    override fun onClick(v: View?) {
    }
    abstract fun clone(): CustomListener

    //dialog that lets users choose intensity of item
    fun chooseIntensity(context: Context, onEnumSelected: (Intensity) -> Unit) {
        val customEnumValues = Intensity.values().copyOfRange(1, Intensity.values().size)
        val customEnumNames = customEnumValues.map { it.name }.toTypedArray()

        AlertDialog.Builder(context)
            .setTitle("Choose Intensity")
            .setItems(customEnumNames) { _, which ->
                onEnumSelected(customEnumValues[which])
            }
            .setNegativeButton("Cancel", null)
            .create()
            .show()
    }

}
